azurerm/resource_arm_application_insights_api_key.go

```diff
@@ -85,7 +85,7 @@ func resourceArmApplicationInsightsAPIKeyCreate(d *schema.ResourceData, meta int
 	resGroup := id.ResourceGroup
 	appInsightsName := id.Path["components"]
 
-	if requireResourcesToBeImported && d.IsNewResource() {
```

Comment:
All the other resources still have ` d.IsNewResource() `, thou as it is a create only function i can see why it is not needed.

Reply:
it's only really required if it's a shared function between Create & Update - I didn't feel it added any value having it here tbh (since there's no way to make this code generic enough to go in a function without sacrificing readability imo?)